### ISSUE TITLE
Wrong needless_pass_by_value for function attributes

### ISSUE THREAD

**Issue by umurgdk** [rust-lang-nursery/rust-clippy#1773]
Clippy confuses function attrs in this example:
```rust
#![feature(plugin)]
#![plugin(rocket_codegen)]

extern crate rocket;

#[get("/")]
fn index() -> &'static str {
    "Hello world!"
}

fn main() {
    rocket::ignite().mount("/", routes![index]).launch();
}
```

Clippy output is:
```
warning: this argument is passed by value, but not consumed in the function body
 --> src/main.rs:6:1
  |
6 | #[get("/")]
  | ^^^^^^^^^^^ help: consider taking a reference instead `&#[get("/")]`
  |
  = note: #[warn(needless_pass_by_value)] on by default
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#needless_pas
s_by_value
```

**Comment by Manishearth** [comment]
It's not confusing functional attrs, the plugin internally expands to a function. This lint needs a macro check

**Comment by Manishearth** [comment]
The lint already has a macro check, looks like it was malfunctioning then (seems to work now)
